protocol/messenger_edit_message_test.go

```diff
@@ -0,0 +1,150 @@
+}
+
+func (s *MessengerEditMessageSuite) TestEditMessage() {
+	theirMessenger := s.newMessenger()
+	_, err := theirMessenger.Start()
+	s.Require().NoError(err)
+
+	theirChat := CreateOneToOneChat("Their 1TO1", &s.privateKey.PublicKey, s.m.transport)
+	err = theirMessenger.SaveChat(theirChat)
+	s.Require().NoError(err)
+
+	ourChat := CreateOneToOneChat("Our 1TO1", &theirMessenger.identity.PublicKey, s.m.transport)
+	err = s.m.SaveChat(ourChat)
+	s.Require().NoError(err)
+
+	inputMessage := buildTestMessage(*theirChat)
+	sendResponse, err := theirMessenger.SendChatMessage(context.Background(), inputMessage)
+	s.NoError(err)
+	s.Require().Len(sendResponse.Messages, 1)
+
+	response, err := WaitOnMessengerResponse(
+		s.m,
+		func(r *MessengerResponse) bool { return len(r.Messages) > 0 },
+		"no messages",
+	)
+	s.Require().NoError(err)
+	s.Require().Len(response.Chats(), 1)
+	s.Require().Len(response.Messages, 1)
+
+	ogMessage := sendResponse.Messages[0]
+
+	editedMessage := buildTestMessage(*theirChat)
+	editedMessage.Replace = ogMessage.ID
+	editedMessage.ContentType = protobuf.ChatMessage_EDIT
+
+	sendResponse, err = theirMessenger.SendChatMessage(context.Background(), editedMessage)
+
+	s.NoError(err)
+	fmt.Println(sendResponse.Messages)
+	s.Require().Len(sendResponse.Messages, 1)
+
+	response, err = WaitOnMessengerResponse(
+		s.m,
+		func(r *MessengerResponse) bool { return len(r.Messages) > 0 },
+		"no messages",
+	)
+	s.Require().NoError(err)
+
+	s.Require().Len(response.Chats(), 1)
+	s.Require().Len(response.Messages, 1)
+	s.Require().Equal(ogMessage.ID, response.Messages[0].Replace)
+
+	// Check if OG message is hidden in the sender side
+	var ogHidden, ogSeen bool
+	err = theirMessenger.persistence.db.QueryRow("SELECT hide, seen FROM user_messages WHERE id = ?", ogMessage.ID).Scan(&ogHidden, &ogSeen)
+	s.Require().NoError(err)
+	s.Require().True(ogHidden)
+	s.Require().True(ogSeen)
+
+	// Check if OG message is hidden in the receiver side
+	err = s.m.persistence.db.QueryRow("SELECT hide, seen FROM user_messages WHERE id = ?", ogMessage.ID).Scan(&ogHidden, &ogSeen)
+	s.Require().NoError(err)
+	s.Require().True(ogHidden)
+	s.Require().True(ogSeen)
+
+	// Main instance user attempts to edit the message it received from theirMessenger
+	editedMessage = buildTestMessage(*ourChat)
+	editedMessage.Replace = ogMessage.ID
+	editedMessage.ContentType = protobuf.ChatMessage_EDIT
+	_, err = s.m.SendChatMessage(context.Background(), editedMessage)
+
+	s.Require().Equal("sender is not the author of the message", err.Error())
+}
+
+func (s *MessengerEditMessageSuite) TestEditMessageOutOfOrder() {
+	theirMessenger := s.newMessenger()
+	_, err := theirMessenger.Start()
+	s.Require().NoError(err)
+
+	theirChat := CreateOneToOneChat("Their 1TO1", &s.privateKey.PublicKey, s.m.transport)
+	err = theirMessenger.SaveChat(theirChat)
+	s.Require().NoError(err)
+
+	inputMessage := buildTestMessage(*theirChat)
+
+	// TODO: send edit before original message
```

Comment:
@Samyoul @cammellos:
For this test I need that `s.m` receives an edited message before receiving the original message. 
How can I achieve this?